File: design-patterns/go/chain_of_responsibility/cache/disk_cache_handler.go
// disk_cache_handler.go

package main

import "fmt"

type DiskCacheHandler struct {
	nextCacheHandler CacheHandler
}

func NewDiskCacheHandler(nextCacheHandler CacheHandler) (diskCacheHandler *DiskCacheHandler) {
	diskCacheHandler = &DiskCacheHandler{}
	diskCacheHandler.nextCacheHandler = nextCacheHandler
	return
}

func (diskCacheHandler *DiskCacheHandler) HandleRequest(data Data) {
	if data.GetDataType() == DATA && len(data.GetValue()) > 1024 {
		fmt.Printf("Caching data '%v' in Disk\n", data.GetKey())
	} else if diskCacheHandler.nextCacheHandler != nil {
		diskCacheHandler.nextCacheHandler.HandleRequest(data)
	}
}

File: design-patterns/php/chainofresponsibility/cache/DiskCacheHandler.php
<?php
// DiskCacheHandler.php

namespace BigBoxCode\DesignPattern\ChainOfResponsibility\Cache;


class DiskCacheHandler extends CacheHandler {
    public function __construct(?CacheHandler $nextCacheHandler) {
        parent::__construct($nextCacheHandler);
    }

    public function handleRequest(Data $data): void {
        if ($data->getType() == DATA_TYPE::DATA && strlen($data->getData()) > 1024) {
            echo "Caching file '" . $data->getKey() . "' in Disk\n";
        } else if ($this->nextCacheHandler != null) {
            $this->nextCacheHandler->handleRequest($data);
        }
    }
}